content/en/tracing/trace_collection/automatic_instrumentation/single-step-apm.md

```diff
@@ -306,6 +306,123 @@ For example, add the following configuration to your `datadog-values.yaml` file:
+   ```shell
+   kubectl create secret generic datadog-secret --from-literal api-key=$DD_API_KEY
+   ```
+2. Create `datadog-agent.yaml` with the spec of your Datadog Agent deployment configuration. The simplest configuration is as follows:
+   ```yaml
+   kind: DatadogAgent
+   apiVersion: datadoghq.com/v2alpha1
+   metadata:
+     name: datadog
+   spec:
+     global:
+       site: <DATADOG_SITE>
+       tags:
+         - env:<AGENT_ENV>
+       credentials:
+         apiSecret:
+           secretName: datadog-secret
+           keyName: api-key
+         appSecret:
+           secretName: datadog-secret
+           keyName: app-key
+   features:
+     apm:
+       instrumentation:
+         enabled: true  
+   ```
+   a. Replace `<DATADOG_SITE>` with your [Datadog site][6].  
+   b. Replace `<AGENT_ENV>` with the environment your Agent is installed on (for example, `env:staging`).
+
+3. Run the following command:
+   ```shell
+   kubectl apply -f /path/to/your/datadog-agent.yaml
+   ```
+4. Restart your applications.
+
+### Enabling or disabling instrumentation for namespaces
+
+You can choose to selectively instrument specific namespaces or choose to not instrument them.
+
+To enable instrumentation for specific namespaces, replace `enabled: true` with `enabledNamespaces` configuration in your `datadog-agent.yaml` file:
+{{< highlight yaml "hl_lines=4-6" >}}
+   features:
+     apm:
+       instrumentation:
+         enabledNamespaces: # Add namespaces to instrument
+           - namespace_1
+           - namespace_2
+ {{< /highlight >}}
+
+<div class="alert alert-info">The <code>enabled: true</code> option enables instrumentation for the entire cluster. You need to remove this to only enable instrumentation for specific namespaces.</a></div>
+
+To disable instrumentation for specific namespaces, add the `disabledNamespaces` configuration to your `datadog-agent.yaml` file:
+{{< highlight yaml "hl_lines=5-7" >}}
+   features:
+     apm:
+       instrumentation:
+         enabled: true
+         enabledNamespaces: # Add namespaces to instrument
+           - namespace_1
+           - namespace_2
+{{< /highlight >}}
+
+### Specifying tracing library versions
+
+You can optionally set specific tracing library versions to use. If you don't specify a version, it defaults to the latest version. To find the latest version for a library, go to **Releases** in the dd-trace-&lt;language&gt; GitHub repo. For example, [dd-trace-dotnet releases][15].
+
+To set specific tracing library versions, add the following configuration to your `datadog-agent.yaml` file:
+{{< highlight yaml "hl_lines=5-10" >}}
+   features:
+     apm:
+       instrumentation:
+         enabled: true
+         libVersions: # Add any versions you want to set
+            dotnet: v2.46.0
+            python: v1.20.6
+            java: v1.22.0
+            js: v4.17.0
+            ruby: v1.15.0
+{{< /highlight >}}
+
+Supported languages include:
```

Comment:
Can we move this further up and move the library version pinning lower. The version pinning would be considered an "advanced" feature.

Actually can we include the language as a prerequisite? "A .Net, Python, Java, ... application to instrument"

Reply:
In a separate PR I'm going to pull the config options into their own section 👍